### PR TITLE
add cargo-spellcheck

### LINKED ISSUE #255: Integrate cargo spellcheck in CI
https://github.com/drahnr/cargo-spellcheck

### PR DESCRIPTION
**Pull request by melekes**
Refs #255

### CODE REVIEW

File: src/libp2p/async_std_connection/with_buffers.rs [hunk 1]
```diff
@@ -92,7 +92,7 @@ impl<T> WithBuffers<T> {
     /// Returns a buffer containing data read from the socket, and a buffer containing data
     /// destined to be written to the socket.
     ///
-    /// If an error happend on the socket earlier, it is returned instead.
+    /// If an error happens on the socket earlier, it is returned instead.
```
Comment (tomaka) [inline src/libp2p/async_std_connection/with_buffers.rs:95]:
```suggestion
    /// If an error happened on the socket earlier, it is returned instead.
```

File: .config/smoldot.dic [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+metadata
+multiaddress/S
+multihash
+mutex/SM
+opcode
+orthogonally
+parachain/S
+parathread/S
+pluggable
+poped
```
Comment (tomaka) [inline .config/smoldot.dic:109]: I think this is a real typo and it should be `popped`, no?

File: .config/smoldot.dic [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+VRF
+Wasi
+Wasm
+WebAssembly
+WebSocket/M
+Westend
+Yamux
+aborter
+allocator
+amongst
```
Comment (tomaka) [inline .config/smoldot.dic:62]: Isn't this a common English word? It's weird

File: src/chain/blocks_tree/verify.rs [hunk 4]
```diff
@@ -770,7 +770,7 @@ pub enum BodyVerifyStep2<T> {
     /// A new runtime must be compiled.
     ///
     /// This variant doesn't require any specific input from the user, but is provided in order to
-    /// make it possible to benchmBodyVerifyStep2<ark the time it takes to compile runtimes.
+    /// make it possible to `benchmBodyVerifyStep2<ark` the time it takes to compile runtimes.
```
Comment (tomaka) [inline src/chain/blocks_tree/verify.rs:773]: Yeah this is clearly me accidentally copy-pasting something

```suggestion
    /// make it possible to benchmark the time it takes to compile runtimes.
```

File: src/chain_spec.rs [hunk 5]
```diff
@@ -291,14 +291,14 @@ impl ChainSpec {
 pub enum Bootnode<'a> {
     /// The address of the bootnode is valid.
     Parsed {
-        /// String representation of the multiaddr that can be used to reach the bootnode.
+        /// String representation of the `Multiaddr` that can be used to reach the bootnode.
```
Comment (tomaka) [inline src/chain_spec.rs:294]:
```suggestion
        /// String representation of the multiaddress that can be used to reach the bootnode.
```

File: src/json_rpc/websocket_server.rs [hunk 6]
```diff
@@ -106,7 +106,7 @@
 //! number of clients (the complexity being `O(n)`).
 //!
 //! The single-threadedness can even be beneficial when the WebSocket server isn't the primary
-//! *raison d'être* of a certain program, which is what this module has been designed for. In the
+//! *`raison d'être`* of a certain program, which is what this module has been designed for. In the
```
Comment (tomaka) [inline src/json_rpc/websocket_server.rs:109]:
```suggestion
//! purpose of a certain program, which is what this module has been designed for. In the
```

https://en.wiktionary.org/wiki/raison_d%27%C3%AAtre

I thought I could be fancy by using a bit of French

File: src/json_rpc/websocket_server.rs [hunk 7]
```diff
@@ -148,7 +148,7 @@ pub struct Config {
 #[derive(Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
 pub struct ConnectionId(usize);
 
-/// WebSockets listening socket and list of open connections.
+/// WebSocket's listening socket and list of open connections.
```
Comment (tomaka) [inline src/json_rpc/websocket_server.rs:151]:
```suggestion
/// Listening socket and list of open connections.
```

It's the plural of "WebSocket". "WebSocket's" doesn't really make sense

**Review by tomaka — CHANGES_REQUESTED** [review]
Thanks :+1:

File: .config/smoldot.dic (hunk 2 above)
Comment (melekes) [reply .config/smoldot.dic]: yep

File: .config/smoldot.dic (hunk 3 above)
Comment (melekes) [reply .config/smoldot.dic:62]: missing here https://github.com/drahnr/cargo-spellcheck/blob/master/hunspell-data/en_US.dic

File: .config/spellcheck.toml [hunk 8]
```diff
@@ -6,9 +6,8 @@ search_dirs = ["."]
 # AFF rules: <https://github.com/drahnr/cargo-spellcheck/blob/master/hunspell-data/en_US.aff>
 extra_dictionaries = ["smoldot.dic"]
 
-# TODO: Uncomment once <https://github.com/drahnr/cargo-spellcheck/issues/15> is done.
-# skip_os_lookups = true
-# use_builtin = true
+skip_os_lookups = true
+use_builtin = true
```
Comment (tomaka) [inline .config/spellcheck.toml:10]: Could you add comments explaining what this is? It's not exactly obvious

Comment (tomaka) [reply .config/spellcheck.toml]: Ah, you did way more than I asked for

**Review by tomaka — APPROVED** [review]

**Comment by tomaka** [comment]
Ah I just realized that this doesn't modify the CI in any way.
I'm ok with just merging the spellcheck fixes, but the issue would remain open.

**Comment by melekes** [comment]
yeah, we'll have to wait until https://github.com/drahnr/cargo-spellcheck/issues/15 is resolved

**Pull request merged**